tests/ui/async-await/async-closures/post-mono-higher-ranked-hang-2.rs

```diff
@@ -0,0 +1,22 @@
+//@ edition: 2021
+//@ check-pass
```

Comment:
this hangs in a check build?

Reply:
whoops, build-pass 💀

Reply:
IIRC it was a cute: check-pass but build-fail

Reply:
whoopsy didn't catch the ci failure